posthog/hogql_queries/web_analytics/test/test_web_stats_table.py

```diff
@@ -1004,3 +1004,52 @@ def test_cohort_test_filters(self):
         ).results
 
         assert results == [["/path1", 1, 1]]
+
+    def test_timezone_filter(self):
+        d1, s1 = "d1", str(uuid7("2024-07-30"))
+        d2, s2 = "d2", str(uuid7("2024-07-30"))
+
+        _create_person(
+            team_id=self.team.pk,
+            distinct_ids=[d1],
+            properties={"name": d1, "email": "test@example.com"},
+        )
+
+        _create_event(
+            team=self.team,
+            event="$pageview",
+            distinct_id=d1,
+            timestamp="2024-07-30",
+            properties={"$session_id": s1, "$pathname": "/path1", "$timezone": "America/New_York"},
+        )
+
+        _create_person(
+            team_id=self.team.pk,
+            distinct_ids=[d2],
+            properties={"name": d2, "email": "d2@hedgebox.net"},
+        )
+        _create_event(
+            team=self.team,
+            event="$pageview",
+            distinct_id=d2,
+            timestamp="2024-07-30",
+            properties={"$session_id": s2, "$pathname": "/path2", "$timezone": "America/Brasilia"},
+        )
+        _create_event(
+            team=self.team,
+            event="$pageview",
+            distinct_id=d2,
+            timestamp="2024-07-30",
+            properties={"$session_id": s2, "$pathname": "/path3", "$timezone": "America/Brasilia"},
+        )
+
+        flush_persons_and_events()
```

Comment:
I don't *think* you need this, I'm not sure if it existed where you copied it from but we could probably remove it there as well.

(I might be wrong, but I think running a query automatically flushes all events)